index.js

```diff
@@ -0,0 +1,175 @@
+      requestedPatient: 'Patient/{{Patient.id}}'
+    }
+  };
+
+  // Example service to invoke the medication-prescribe hook
+  const medicationPrescribeExample = {
+    hook: 'medication-prescribe',
+    id: 'medication-prescribe-example',
+    title: 'Example medication-prescribe CDS Service',
+    description: 'Suggests prescribing Aspirin 81 MG Oral Tablets',
+  };
+
+  const discoveryEndpointServices = {
+    services: [ patientViewExample, medicationPrescribeExample ]
+  };
+  response.json(discoveryEndpointServices);
+});
+
+/**
+ * Patient View Example Service:
+ * - Handles POST requests to our patient-view-example endpoint
+ * - This function should respond with an array of card(s) in JSON format for the patient-view hook
+ *
+ * - Service purpose: Display a patient's first and last name, with a link to the CDS Hooks web page
+ */
+app.post('/cds-services/patient-view-example', (request, response) => {
+
+  // Parse the request body for the Patient prefetch resource
+  const patientResource = request.body.prefetch.requestedPatient.resource;
+  const patientViewCard = {
+    cards: [
+      {
+        // Use the patient's First and Last name
+        summary: 'Now seeing: ' + patientResource.name[0].given[0] + ' ' + patientResource.name[0].family[0],
+        indicator: 'info',
+        links: [
+          {
+            label: 'Learn more about CDS Hooks',
+            url: 'http://cds-hooks.org',
+            type: 'absolute'
+          }
+        ]
+      }
+    ]
+  };
+  response.json(patientViewCard);
+});
+
+/**
+ * Medication Prescribe Example Service:
+ * - Handles POST requests to the medication-prescribe-example endpoint
+ * - This function should respond with an array of cards in JSON format for the medication-prescribe hook
+ *
+ * - Service purpose: Upon a provider choosing a medication to prescribe, display a suggestion for the
+ *                    provider to change their chosen medication to the service-recommended Aspirin 81 MG Oral Tablet,
+ *                    or display text that affirms the provider is currently prescribing the service-recommended Aspirin
+ */
+app.post('/cds-services/medication-prescribe-example', (request, response) => {
+
+  // Parse the request body for the FHIR context provided by the EHR. In this case, the MedicationOrder resource
+  const context = request.body.context[0];
+
+  // Check if a medication was chosen by the provider to be ordered
+  if (context.medicationCodeableConcept) {
+    const responseCard = createResponseCard(context); // see function below for more details
+    response.json(responseCard);
+  }
+  response.status(200);
+});
+
+/**
+ * Creates a Card array based upon the medication chosen by the provider in the request context
+ * @param context - The FHIR context of the medication being ordered by the provider
+ * @returns {{cards: *[]}} - Either a card with the suggestion to switch medication or a textual info card
+ */
+function createResponseCard(context) {
```

Comment:
this is only for medication right?`createMedicationResponseCard`?

Reply:
Correct, just to abstract a large chunk of code from the POST block. I'll change the name accordingly.